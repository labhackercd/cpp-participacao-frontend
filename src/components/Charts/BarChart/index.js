import React from 'react';
import Chart from "react-google-charts";

export default function BarChart(props) {
  return (
    <Chart
      width={'500px'}
      height={'300px'}
      chartType="BarChart"
      loader={<div>Loading Chart</div>}
      data={props.data}
      options={props.options}
      rootProps={{ 'data-testid': 'bar-chart' }}
    />
   );
}
