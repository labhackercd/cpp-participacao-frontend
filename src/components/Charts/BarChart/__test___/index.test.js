import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from "enzyme";
import BarChart from './../index';

const barChartdata = {
    values: [
    ['City', '2010 Population', '2000 Population'],
    ['New York City, NY', 8175000, 8008000],
    ['Los Angeles, CA', 3792000, 3694000],
    ['Chicago, IL', 2695000, 2896000],
    ['Houston, TX', 2099000, 1953000],
    ['Philadelphia, PA', 1526000, 1517000],
  ],
    options:{
    title: 'Population of Largest U.S. Cities',
    chartArea: { width: '50%' },
    hAxis: {
      title: 'Total Population',
      minValue: 0,
    },
    vAxis: {
      title: 'City',
    },
  }
};

test("Snapshot should not have changes", () => {
    const component = mount(<BarChart data={barChartdata.values} options={barChartdata.options}></BarChart>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

it("Mounts the chart with the data", () => {
    const div = document.createElement("div")
    ReactDOM.render(<BarChart data={barChartdata.values} options={barChartdata.options}></BarChart>, div)
});
