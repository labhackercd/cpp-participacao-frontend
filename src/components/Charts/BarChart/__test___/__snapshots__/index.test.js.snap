// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should not have changes 1`] = `
<BarChart
  data={
    Array [
      Array [
        "City",
        "2010 Population",
        "2000 Population",
      ],
      Array [
        "New York City, NY",
        8175000,
        8008000,
      ],
      Array [
        "Los Angeles, CA",
        3792000,
        3694000,
      ],
      Array [
        "Chicago, IL",
        2695000,
        2896000,
      ],
      Array [
        "Houston, TX",
        2099000,
        1953000,
      ],
      Array [
        "Philadelphia, PA",
        1526000,
        1517000,
      ],
    ]
  }
  options={
    Object {
      "chartArea": Object {
        "width": "50%",
      },
      "hAxis": Object {
        "minValue": 0,
        "title": "Total Population",
      },
      "title": "Population of Largest U.S. Cities",
      "vAxis": Object {
        "title": "City",
      },
    }
  }
>
  <Chart
    chartActions={null}
    chartEvents={null}
    chartType="BarChart"
    chartWrapperParams={Object {}}
    className=""
    columns={null}
    controls={null}
    data={
      Array [
        Array [
          "City",
          "2010 Population",
          "2000 Population",
        ],
        Array [
          "New York City, NY",
          8175000,
          8008000,
        ],
        Array [
          "Los Angeles, CA",
          3792000,
          3694000,
        ],
        Array [
          "Chicago, IL",
          2695000,
          2896000,
        ],
        Array [
          "Houston, TX",
          2099000,
          1953000,
        ],
        Array [
          "Philadelphia, PA",
          1526000,
          1517000,
        ],
      ]
    }
    diffdata={null}
    formatters={null}
    getChartEditor={null}
    getChartWrapper={[Function]}
    graphID={null}
    graph_id={null}
    height="300px"
    legendToggle={false}
    legend_toggle={false}
    loader={
      <div>
        Loading Chart
      </div>
    }
    options={
      Object {
        "chartArea": Object {
          "width": "50%",
        },
        "hAxis": Object {
          "minValue": 0,
          "title": "Total Population",
        },
        "title": "Population of Largest U.S. Cities",
        "vAxis": Object {
          "title": "City",
        },
      }
    }
    render={null}
    rootProps={
      Object {
        "data-testid": "bar-chart",
      }
    }
    rows={null}
    spreadSheetQueryParameters={
      Object {
        "gid": 1,
        "headers": 1,
      }
    }
    spreadSheetUrl={null}
    style={Object {}}
    toolbarID={null}
    toolbarItems={null}
    width="500px"
  >
    <ContextProvider
      value={
        Object {
          "chartActions": null,
          "chartEvents": null,
          "chartType": "BarChart",
          "chartWrapperParams": Object {},
          "className": "",
          "columns": null,
          "controls": null,
          "data": Array [
            Array [
              "City",
              "2010 Population",
              "2000 Population",
            ],
            Array [
              "New York City, NY",
              8175000,
              8008000,
            ],
            Array [
              "Los Angeles, CA",
              3792000,
              3694000,
            ],
            Array [
              "Chicago, IL",
              2695000,
              2896000,
            ],
            Array [
              "Houston, TX",
              2099000,
              1953000,
            ],
            Array [
              "Philadelphia, PA",
              1526000,
              1517000,
            ],
          ],
          "diffdata": null,
          "formatters": null,
          "getChartEditor": null,
          "getChartWrapper": [Function],
          "graphID": null,
          "graph_id": null,
          "height": "300px",
          "legendToggle": false,
          "legend_toggle": false,
          "loader": <div>
            Loading Chart
          </div>,
          "options": Object {
            "chartArea": Object {
              "width": "50%",
            },
            "hAxis": Object {
              "minValue": 0,
              "title": "Total Population",
            },
            "title": "Population of Largest U.S. Cities",
            "vAxis": Object {
              "title": "City",
            },
          },
          "render": null,
          "rootProps": Object {
            "data-testid": "bar-chart",
          },
          "rows": null,
          "spreadSheetQueryParameters": Object {
            "gid": 1,
            "headers": 1,
          },
          "spreadSheetUrl": null,
          "style": Object {},
          "toolbarID": null,
          "toolbarItems": null,
          "width": "500px",
        }
      }
    >
      <div>
        Loading Chart
      </div>
      <GoogleChartLoader
        onError={[Function]}
        onLoad={[Function]}
      >
        <Script
          attributes={Object {}}
          onCreate={[Function]}
          onError={[Function]}
          onLoad={[Function]}
          url="https://www.gstatic.com/charts/loader.js"
        />
      </GoogleChartLoader>
    </ContextProvider>
  </Chart>
</BarChart>
`;
